pyrsa/rdm/rdms.py

```diff
@@ -374,6 +377,130 @@ def sort_by(self, **kwargs):
+        n_rdms = sum([rdms.n_rdm for rdms in list_of_rdms])
+        n_patterns = len(all_patterns)
+
+        desc_tuples = []
+        rdm_desc_names = []
+        for rdms in list_of_rdms:
+            desc_tuples += list(rdms.descriptors.items())
+            rdm_desc_names += list(rdms.rdm_descriptors.keys())
+        descriptors = dict(
+            [d for d, c in Counter(desc_tuples).items() if c == n_rdm_objs]
+        )
+        desc_diff_names = set(
+            [d[0] for d, c in Counter(desc_tuples).items() if c != n_rdm_objs]
+        )
+        rdm_desc_names = set(rdm_desc_names + list(desc_diff_names))
+        rdm_descriptors = dict([(n, [None]*n_rdms) for n in rdm_desc_names])
+
+        measure = None
+        vector_len = int(n_patterns * (n_patterns-1) / 2)
+        vectors = np.full((n_rdms, vector_len), np.nan)
+        rdm_id = 0
+        for rdms in list_of_rdms:
+            measure = rdms.dissimilarity_measure
+            pidx = [all_patterns.index(i) for i in pdescs(rdms, descriptor)]
+            for rdm_local_id, utv in enumerate(rdms.dissimilarities):
+                rdm = np.full((len(all_patterns), len(all_patterns)), np.nan)
+                rdm[np.ix_(pidx, pidx)] = squareform(utv, checks=False)
+                vectors[rdm_id, :] = squareform(rdm, checks=False)
+                for name in rdm_descriptors.keys():
+                    if name in rdms.rdm_descriptors:
+                        val = rdms.rdm_descriptors[name][rdm_local_id]
+                        rdm_descriptors[name][rdm_id] = val
+                    elif name in rdms.descriptors:
+                        rdm_descriptors[name][rdm_id] = rdms.descriptors[name]
+                rdm_id += 1
+
+        return RDMs(
+            dissimilarities=vectors,
+            dissimilarity_measure=measure,
+            descriptors=descriptors,
+            rdm_descriptors=rdm_descriptors,
+            pattern_descriptors=dict([(descriptor, all_patterns)])
+        )
+
+    def align(self, method='evidence'):
+        """Bring RDMs closer together
+
+        Iteratively scales RDMs based on pairs in-common.
+        Also adds an RDM descriptor with the weights used.
+
+        Args:
+            method (str, optional): One of 'evidence', 'setsize' or
+                'simple'. Defaults to 'evidence'.
+
+        Returns:
+            RDMs: RDMs object with the aligned RDMs
+        """
+        aligned, weights = _align(self.dissimilarities, method)
+        rdm_descriptors = deepcopy(self.rdm_descriptors)
+        if weights is not None:
+            rdm_descriptors['weights'] = weights
+        return RDMs(
+            dissimilarities=aligned,
+            dissimilarity_measure=self.dissimilarity_measure,
+            descriptors=deepcopy(self.descriptors),
+            rdm_descriptors=rdm_descriptors,
+            pattern_descriptors=deepcopy(self.pattern_descriptors)
+        )
+
+    def mean(self, weights='stored'):
```

Comment:
I find the default of using stored weights a bit odd here.  I would certainly prefer the None version to be the default, i.e. the version which takes an unweighted average. rdms.mean() should really work. Also I am not sure, why self.descriptors['weights'] has to be removed. 

Would a version which accepts the name of the weights descriptor as input make sense? Such that the current behavior could be achieved by rdms.mean('weights') instead?